productivity-suite/scripts/start-dev.js

```diff
@@ -37,31 +37,26 @@ async function runAllProcesses() {
 
     await sleep(5000);
 
-    log('Serving the legacy application...');
-    runNpmScript(path.join(appDir, 'legacy-app'), 'dev.react');
-    runNpmScript(path.join(appDir, 'legacy-app'), 'dev.worker');
-
-    await sleep(2000);
-
     const fragmentsDirPath = path.join(rootDir, 'app', 'fragments');
-
     const fragments = fs.readdirSync(fragmentsDirPath, { withFileTypes: true })
         .filter(file => file.isDirectory())
-        .map(file => ({
-            name: file.name,
-            path: path.join(fragmentsDirPath, file.name)
-        }));
 
     for(const fragment of fragments) {
         log(`Serving the ${fragment.name} fragment...`);
-		runNpmScript(fragment.path);
-		await sleep(3000);
+		runNpmScript(path.join(fragmentsDirPath, fragment.name));
+		await sleep(2000);
 	}
 
-    await sleep(3000);
+    log('Serving the legacy application...');
+    runNpmScript(path.join(appDir, 'legacy-app'), 'dev.react');
+
+    await sleep(2000);
+
+    log(`Serving the Gateway worker...`);
+    runNpmScript(path.join(appDir, 'legacy-app'), 'dev.worker');
```

Comment:
are you sure this is ok? I remember specifically starting the gateway worker first because otherwise it seemed like it would not recognise the bindings (or something like that) 🤔

Reply:
I've just tested it locally and it seems to work fine... maybe something has changed... or maybe I did something wrong back then 🤷

Reply:
It should not matter too much since the way Wrangler dev mode works, it proxies the service-bindings anyway. But... in general, it is safer to run the dependencies first. In the case of deployment (or remote dev mode) you absolutely need to have the dependencies uploaded before the depending workers.